pages/docs/typescript.en-US.mdx

```diff
@@ -0,0 +1,76 @@
+# TypeScript
+
+SWR is friendly for apps which are written in TypeScript. It's type safe out of the box.
+
+### Basic Usage
+
+By default, SWR will also infer the type of `key` argument in `fetcher`. So you can use the preferred types of `key` and `fetcher`.
+
+#### Basic hook
+
+```typescript
+// `key` is inferred to be `string`
+useSWR('/api/user', key => {})
+// `key` will be inferred as { a: string; b: { c: string; d: number } }
+useSWR({ a: '1', b: { c: '3', d: 2 } }, key => {})
+```
+
+You can explicitly the types for key and fetcher arguments.
+
+```typescript
+import useSWR, { Key, Fetcher } from 'swr'
+
+
+const key: Key = 'string'
+const fetcher: Fetcher<string, string> = (k) => k + ':value'
+useSWR(key, fetcher)
+```
+
+
+#### Inifinite
+
+It's same for `swr/inifite`, you can either rely on the automatically inferring or explicitly sepicify the types.
+
+```typescript
+import { SWRInfiniteKeyLoader } from 'swr/infinite'
+
+const keyLoader: SWRInfiniteKeyLoader = (index, previousPageData) => { /* ... */ };
+const useList = () => {
+  const { data } = useSWRInfinite(keyLoader, fetcher);
+  // ...
+};
+```
+
+### Generics
+
+Speicifying the type of `data` is easy.
```

Comment:
```suggestion
Speicifying the type of `data` is easy. By default, it will use the return type of `fetcher` (with `undefined` for the non-ready state) as the `data` type, but you can also pass it as a parameter:
```